server/src/main/java/org/elasticsearch/script/ScriptStats.java

```diff
@@ -9,46 +9,37 @@
 package org.elasticsearch.script;
 
 import org.elasticsearch.TransportVersion;
+import org.elasticsearch.common.collect.Iterators;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.common.io.stream.Writeable;
 import org.elasticsearch.common.util.Maps;
-import org.elasticsearch.xcontent.ToXContentFragment;
-import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.ChunkedToXContent;
+import org.elasticsearch.common.xcontent.ChunkedToXContentHelper;
+import org.elasticsearch.xcontent.ToXContent;
 
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Collections;
+import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
-
-public class ScriptStats implements Writeable, ToXContentFragment {
-    private final List<ScriptContextStats> contextStats;
-    private final long compilations;
-    private final long cacheEvictions;
-    private final long compilationLimitTriggered;
-    private final TimeSeries compilationsHistory;
-    private final TimeSeries cacheEvictionsHistory;
-
-    public ScriptStats(List<ScriptContextStats> contextStats) {
-        ArrayList<ScriptContextStats> ctxStats = new ArrayList<>(contextStats.size());
-        ctxStats.addAll(contextStats);
-        ctxStats.sort(ScriptContextStats::compareTo);
-        this.contextStats = Collections.unmodifiableList(ctxStats);
-        long compilations = 0;
-        long cacheEvictions = 0;
-        long compilationLimitTriggered = 0;
-        for (ScriptContextStats stats : contextStats) {
-            compilations += stats.getCompilations();
-            cacheEvictions += stats.getCacheEvictions();
-            compilationLimitTriggered += stats.getCompilationLimitTriggered();
-        }
-        this.compilations = compilations;
-        this.cacheEvictions = cacheEvictions;
-        this.compilationLimitTriggered = compilationLimitTriggered;
-        this.compilationsHistory = new TimeSeries(compilations);
-        this.cacheEvictionsHistory = new TimeSeries(cacheEvictions);
-    }
+import java.util.Objects;
+
+import static org.elasticsearch.common.collect.Iterators.single;
+import static org.elasticsearch.script.ScriptContextStats.Fields.COMPILATIONS_HISTORY;
+import static org.elasticsearch.script.ScriptStats.Fields.CACHE_EVICTIONS;
+import static org.elasticsearch.script.ScriptStats.Fields.COMPILATIONS;
+import static org.elasticsearch.script.ScriptStats.Fields.COMPILATION_LIMIT_TRIGGERED;
+import static org.elasticsearch.script.ScriptStats.Fields.CONTEXTS;
+import static org.elasticsearch.script.ScriptStats.Fields.SCRIPT_STATS;
+
+public record ScriptStats(
```

Comment:
Nit: I understand this was here before you touched it, but if you have a good view of what is this class doing, do you mind adding a javadoc?